### PR TITLE
docs: record distroless images

### PR DESCRIPTION
**Pull request by daixiang0**
Signed-off-by: Long Dai <long0dai@foxmail.com>

Commit Message:

Document addition of distroless Docker image:

- add to table of available docker images
- add in getting started > install > docker (in a separate tab - Envoy (distroless))

Additional Description:
Risk Level: Low
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Partial fix for #16282

### CODE REVIEW

**Comment by phlax** [comment]
docs are here https://storage.googleapis.com/envoy-pr/16359/docs/index.html

File: docs/root/start/install.rst [hunk 1]
```diff
@@ -249,6 +256,18 @@ The following table shows the available Docker images
      -
      - See Docker Hub
      - See Docker Hub
+   * - `envoyproxy/envoy-distroless <https://hub.docker.com/r/envoyproxy/envoy-distroless/tags/>`_
```
Comment (phlax) [inline docs/root/start/install.rst:259]: this needs to be above `envoy-alpine`

File: docs/root/start/install.rst [hunk 2]
```diff
@@ -249,6 +256,18 @@ The following table shows the available Docker images
...
+     - Build image on top of distroless images.
```
Comment (phlax) [inline docs/root/start/install.rst:260]: this text is wrong use the alpine equivalent for example

File: docs/root/start/install.rst [hunk 3]
```diff
@@ -249,6 +256,18 @@ The following table shows the available Docker images
...
+     -
+     -
+     - See Docker Hub
+     - See Docker Hub
+   * - `envoyproxy/envoy-distroless-dev <https://hub.docker.com/r/envoyproxy/envoy-distroless-dev/tags/>`_
+     - Build image on top of distroless images.
```
Comment (phlax) [inline docs/root/start/install.rst:266]: same as above - move up and fix text

**Comment by daixiang0** [comment]
ping @phlax

**Review by phlax — APPROVED** [review]
hi @daixiang0 looking good

the image table needs some further work

File: docs/root/start/install.rst [hunk 4]
```diff
@@ -201,6 +208,18 @@ The following table shows the available Docker images
...
      -
      -
+   * - `envoyproxy/envoy-distroless <https://hub.docker.com/r/envoyproxy/envoy-distroless/tags/>`_
+     - Release binary with symbols stripped on top of a distroless Debain base.
+     -
+     -
+     - See Docker Hub
+     - See Docker Hub
+   * - `envoyproxy/envoy-distroless-dev <https://hub.docker.com/r/envoyproxy/envoy-distroless-dev/tags/>`_
+     - Release binary with symbols stripped on top of a distroless Debain base.
```
Comment (phlax) [inline docs/root/start/install.rst:218]: move this down - so it matches existing order - ie dev images are after

also Debian is spelt incorrectly - but i think rather remove it altogether as its confusing here and not needed - so just a "distroless base"

**Review by phlax — CHANGES_REQUESTED** [review]
oops approved and didnt add change request...

File: docs/root/start/install.rst (hunk 4 above)
Comment (daixiang0) [reply docs/root/start/install.rst:218]: Got it

File: docs/root/start/install.rst [hunk 5]
```diff
@@ -201,6 +208,12 @@ The following table shows the available Docker images
      - |DOCKER_IMAGE_TAG_NAME|
      -
      -
+   * - `envoyproxy/envoy-distroless <https://hub.docker.com/r/envoyproxy/envoy-distroless/tags/>`_
+     - Release binary with symbols stripped on top of a distroless base.
+     -
+     -
+     - See Docker Hub
```
Comment (phlax) [inline docs/root/start/install.rst:215]: please also make these match the alpine images (ie amd64 only and the correct version info)

Comment (daixiang0) [reply docs/root/start/install.rst]: Could I use `DOCKER_IMAGE_TAG_NAME` directly?

Comment (phlax) [reply docs/root/start/install.rst]: that is correct

**Comment by phlax** [comment]
@daixiang0 looks good

while testing out the docs i realised that there are some other problems that need to be fixed first - the existing instructions dont work correcctly - and this makes it worse

ill create a PR to fix docs shortly...

**Review by phlax — APPROVED** [review]
@daixiang0 i created 2 prs to address issues uncovered while testing your PR

one was an ongoing issue in relation to the docker docs #16380 and the other related to the distroless image in particular #16383 

im happy for this to land in the meantime

**Comment by daixiang0** [comment]
Thanks a lot!

**Pull request merged**